connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -253,13 +256,30 @@ public ToleranceType getErrorToleranceType() {
         return errorToleranceType;
     }
 
-    // Visible for testing
+    /**
+     * Check whether we can continue retrying or not.
+     * Visible for testing.
+     * @param startTime the time in milliseconds when the retriable operation was first begun
+     * @return true if we can continue retrying; false if the retry timeout has been reached and we can't retry anymore
+     */
     boolean checkRetry(long startTime) {
+        if (errorRetryTimeout < 0) {
+            return true;
+        }
         return (time.milliseconds() - startTime) < errorRetryTimeout;
```

Comment:
This can be simplified:

```suggestion
        return errorRetryTimeout < 0
                || (time.milliseconds() - startTime) < errorRetryTimeout;
```